### PR TITLE
GEODE-7927: create tests to ensure feature-parity with Redis PEXPIRE …

### PR DESCRIPTION
**Pull request by jhutchison**
…command

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  private static GeodeRedisServer server;
+  private static GemFireCache cache;
+  private static int port = 6379;
+
+  @BeforeClass
+  public static void setUp() {
+    port = AvailablePortHelper.getRandomAvailableTCPPort();
+    server = new GeodeRedisServer("localhost", port);
+    server.start();
+    jedis = new Jedis("localhost", port, 10000000);
```
Comment (moleske) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java]: `10000000` is a timeout I think?  I saw in `PubSubIntegrationTest` the `10000000` was assigned to a constant of `REDIS_CLIENT_TIMEOUT` which would make it a little clearer without diving into the constructor (thought Intellij tells me what the parameter is so not too worried about it)

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+
+  @AfterClass
+  public static void tearDown() {
+    jedis.close();
+    server.shutdown();
+  }
+
+  @Test
+  public void Should_SetExpiration_givenKeyTo_StringValueInMilliSeconds() {
```
Comment (moleske) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java]: Thank you for a very clear test name!

File: geode-redis/src/test/java/org/apache/geode/redis/internal/executor/general/PexpireExecutorJUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.apache.geode.redis.internal.Executor;
+import org.apache.geode.redis.internal.executor.ExpireExecutor;
+import org.apache.geode.redis.internal.executor.PExpireExecutor;
+
+public class PexpireExecutorJUnitTest {
+
+  @Test
+  public void isAnInstanceOfExpireClass() {
+    Executor subject = new PExpireExecutor();
```
Comment (moleske) [inline geode-redis/src/test/java/org/apache/geode/redis/internal/executor/general/PexpireExecutorJUnitTest.java]: I'm sure this test is important, but it seems odd that there's nothing else to test other than it is an `instanceof`

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import org.apache.geode.cache.GemFireCache;
+import org.apache.geode.internal.AvailablePortHelper;
+import org.apache.geode.redis.GeodeRedisServer;
+
+public class PexpireIntegrationTest {
+
+  private static Jedis jedis;
+  private static GeodeRedisServer server;
+  private static GemFireCache cache;
+  private static int port = 6379;
```
[the remark this answers is not in the document]
Comment (jhutchison) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java]: When will I learn - copy/paste is not my friend ?!?  Point taken...

File: geode-redis/src/test/java/org/apache/geode/redis/internal/executor/general/PexpireExecutorJUnitTest.java (hunk 3 above)
Comment (jhutchison) [reply geode-redis/src/test/java/org/apache/geode/redis/internal/executor/general/PexpireExecutorJUnitTest.java]: Not sure how to make this more clear- this idea is that there is a base class (expireExecutor) where 99% of functionality happens.   The Pexpireexecutor class calls uses the default base class for all functionality except for converting the seconds paramater to milliseconds.  SO we tests the millisecond functionality in the  extended integration test for the pexpire class, and test that it is extending its parent.  I was a little unsure about this one, but it didn;t seem that it was worth having all of the other unit tests to test the underlying base class (again).  Suggestions welcome...

Comment (moleske) [reply geode-redis/src/test/java/org/apache/geode/redis/internal/executor/general/PexpireExecutorJUnitTest.java]: Oh the test is fine! I meant it was just weird to me that there wasn't anything else to test.  Sorry for the confusion!  Thank you for the clarifying response explaining that all the cool stuff happens in the base class

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (jhutchison) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java]: corrected.  Thanks for the call out

[the remark this answers is not in the document]
Comment (jhutchison) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java]: (oops sorry-  not pushed yet... my over-eager fingers)

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java (hunk 1 above)
Comment (jhutchison) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PexpireIntegrationTest.java]: corrected-  thx

**Pull request merged**